### DISCUSSION TITLE
When parsing an unknown enum value, set a default (or None)

### DISCUSSION THREAD

**Discussion by br-follow** [pydantic/pydantic#4417] [Question] [answered]
I am parsing JSON responses into pydantic models, some of which have enum fields. When the response contains an unrecognized value for an enum, the parsing fails. 

How can I set the pydantic model's behavior to allow for an unknown enum value and set a default value instead? Additionally, the ability to log (or store) the raw value would be useful has well.

**Comment by samuelcolvin** [comment, marked as answer]
See [custom data types](https://pydantic-docs.helpmanual.io/usage/types/#custom-data-types).

add `__get_validators__` to your enum and implement your own logic from there.
